tests/vehicle_drag.cpp

```diff
@@ -0,0 +1,289 @@
+            MAPSIZE * SEEY, 0 ) ) ) {
+        g->m.destroy_vehicle( veh.v );
+    }
+
+    g->m.build_map_cache( 0, true );
+}
+
+
+vehicle *setup_drag_test( const vproto_id &veh_id )
+{
+    clear_game_drag( ter_id( "t_pavement" ) );
+
+    const tripoint map_starting_point( 60, 60, 0 );
+    vehicle *veh_ptr = g->m.add_vehicle( veh_id, map_starting_point, -90, 100, 0 );
+
+    REQUIRE( veh_ptr != nullptr );
+    if( veh_ptr == nullptr ) {
+        return nullptr;
+    }
+
+    // start the engines, close the doors
+    veh_ptr->start_engines();
+    const auto doors = veh_ptr->get_avail_parts( "OPENABLE" );
+    for( const vpart_reference vp :  doors ) {
+        const size_t door = vp.part_index();
+        veh_ptr->close( door );
+    }
+
+    veh_ptr->refresh_insides();
+    return veh_ptr;
+}
+
+// Algorithm goes as follows:
+// Clear map
+// Spawn a vehicle
+// calculate c_air_drag and c_rolling_resistance
+// return whether they're within 5% of expected values
+bool test_drag( const vproto_id &veh_id,
+                const double expected_c_air, const double expected_c_rr,
+                const int expected_safe, const int expected_max )
+{
+    vehicle *veh_ptr = setup_drag_test( veh_id );
+    if( veh_ptr == nullptr ) {
+        return false;
+    }
+
+    double c_air = veh_ptr->coeff_air_drag();
+    double c_rolling = veh_ptr->coeff_rolling_drag();
+    int safe_v = veh_ptr->safe_velocity();
+    int max_v = veh_ptr->max_velocity();
+
+    const auto d_in_bounds = [&]( double expected, double value ) {
+        double expected_high = expected * 1.05;
+        double expected_low = expected * 0.95;
+        CHECK( value >= expected_low );
+        CHECK( value <= expected_high );
+        return ( value >= expected_low ) && ( value <= expected_high );
+    };
+    const auto i_in_bounds = [&]( int expected, int value ) {
+        int expected_high = expected * 1.05;
+        int expected_low = expected * 0.95;
+        CHECK( value >= expected_low );
+        CHECK( value <= expected_high );
+        return ( value >= expected_low ) && ( value <= expected_high );
+    };
+    bool valid = true;
+    valid &= d_in_bounds( expected_c_air, c_air );
+    valid &= d_in_bounds( expected_c_rr, c_rolling );
+    valid &= i_in_bounds( expected_safe, safe_v );
+    valid &= i_in_bounds( expected_max, max_v );
+    return valid;
+}
+
+// Behold: power of laziness
+void print_drag_test_strings( const std::string &type )
+{
+    std::ostringstream ss;
+    vehicle *veh_ptr = setup_drag_test( vproto_id( type ) );
+    if( veh_ptr == nullptr ) {
+        return;
+    }
+    ss << "    test_vehicle_drag( \"" << type << "\", ";
+    ss << veh_ptr->coeff_air_drag() << ", ";
+    ss << veh_ptr->coeff_rolling_drag() << ", ";
+    ss << veh_ptr->safe_velocity() << ", ";
+    ss << veh_ptr->max_velocity();
+    ss << " );" << std::endl;
+    printf( "%s", ss.str().c_str() );
```

Comment:
Have you considered printing this directly to `std::cout`? Why the separate string stream?